### PR TITLE
Add Clean.unicodeDefault (AIDH-1863)

### PR DESCRIPTION
**Pull request by jdcba**
v2.27.0 adds optional support for Unicode character handling when `Clean`ing `Field`s. Adds missing unit tests.

This extends Maestro in direct analogy to its existing POSIX US-ASCII charset handling: preserves Printable characters only (effectively trims leading/trailing whitespace & filters out control characters).

Leverages Unicode TR#18 Annex C support in Java’s `Pattern`s as the _de facto_ acceptable character policy (see also https://github.com/CommBank/OmniaDesignForum/blob/289dc0331087ac7ee90f6957a0d2c6107c9106fe/features/string-features.md).

This is intended as a stepping stone for #425 because we have an in-flight job requirement.

### CODE REVIEW

File: maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/clean/Clean.scala [hunk 1]
```diff
@@ -30,11 +30,23 @@ object Clean {
   def trim: Clean =
     Clean((_, data) => data.trim)
 
+  /** Keep POSIX US-ASCII printables only.
+    * @see https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html#posix
```
Comment (rowandavies) [inline maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/clean/Clean.scala]: Nice.  I was just about to ask for a reference on this so I could review quicker.

File: maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/clean/CleanSpec.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  val allAscii = (0 to 127).map(_.toChar).mkString
+
+  val posixPrintable =
+    " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~"
+
+  val isPosixPrintable = posixPrintable.toSet
+
+  def isUnicodePrintable(c: Char): Boolean =
+    c.toString.matches("(?U)[\\p{Graph}\\p{Blank}&&[^\\p{Cntrl}]]")
```
Comment (rowandavies) [inline maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/clean/CleanSpec.scala]: Can you teach me quickly in person so I'm confident that this pattern tests correctly?  Or else a second pair of eyes would be sufficient.

Comment (rowandavies) [inline maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/clean/CleanSpec.scala]: Doh!  Andy beat me to it.  Never mind.

Comment (jdcba) [inline maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/clean/CleanSpec.scala]: The Scaladoc references http://www.unicode.org/reports/tr18/#Compatibility_Properties which says: `\p{graph} \p{blank} -- \p{cntrl}` and the Scaladoc also references https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html#posix which says “\p{Print}	A printable character: [\p{Graph}\p{Blank}&&[^\p{Cntrl}]]” and “when UNICODE_CHARACTER_CLASS flag is specified” and “The UNICODE_CHARACTER_CLASS mode can also be enabled via the embedded flag expression (?U).” Therefore, putting it all together, I came up with line 44.

**Comment by jdcba** [comment]
Thanks guys! Merging...

**Pull request merged**